src/main/java/edu/project2/tripplanner/service/CommentServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+package edu.project2.tripplanner.service;
+
+import edu.project2.tripplanner.dto.CommentDTO;
+import edu.project2.tripplanner.dto.CommentIdDTO;
+import edu.project2.tripplanner.exception.Message;
+import edu.project2.tripplanner.exception.NotFoundException;
+import edu.project2.tripplanner.model.Comment;
+import edu.project2.tripplanner.repository.CommentRepository;
+import edu.project2.tripplanner.repository.PlaceRepository;
+import edu.project2.tripplanner.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class CommentServiceImpl implements CommentService, Message {
+    private final CommentRepository commentRepository;
+    private final  UserRepository userRepository;
+    private final PlaceRepository placeRepository;
+
+    private Long userId;
+    private Long placeId;
+
+    @Override
+    public List<Comment> findByUserId(Long userId) {
+
+        return commentRepository.findByUserId(userId);
+    }
+
+    @Override
+    public Optional<Comment> findByIdAndUserId(Long id, Long userId) {
+
+        return commentRepository.findByIdAndUserId(id, userId);
+    }
+
+    @Override
+    public Optional<Comment> findByIdAndUserIdAndPlaceId(Long id, Long userId, Long placeId) {
+
+        return commentRepository.findByIdAndUserIdAndPlaceId(id, userId, placeId);
+    }
+
+    @Override
+    public void deleteCommentById(CommentIdDTO commentIdDTO) {
+         userId=commentIdDTO.getUserId();
+        Long commentId=commentIdDTO.getCommentId();
+         placeId=commentIdDTO.getPlaceId();
+         commentRepository.findByIdAndUserIdAndPlaceId(commentId, userId, placeId).map(c -> {
+            commentRepository.deleteById(commentId);
+
+            return ResponseEntity.ok().build();
```

Comment:
no...
it's bad
You could write it like that: 

commentRepository.findByIdAndUserIdAndPlaceId(commentDTO.getCommentId(), commentDTO.getUserId(), commentDTO.getPlaceId).ifPresent(comment -> commentRepository.deleteById()

I'm actually not a fan of this style with ResponseEntity. In this case, you depend on this. What if I want to call this method from another service for example? I don't need this ResponseEntity - it's a part of controller, not a service. Why your service should return smth, which related to controller staff?
It this case, for delete, just leave it void. And in controller build response and return it FROM controller.

Your method should be more flexible. Plus, remember Single responsibility principle. If your method removes smth, why it should return smth, not related to removing? (ResponseEntity)

Reply:
And one more thing. You could create a custom query for removing in repository
Smth like that:  void deleteByIdAndUserIdAndPlaceId(params...)